lib/core/src/sdk.rs

```diff
@@ -676,23 +676,50 @@ impl LiquidSdk {
     ) -> Result<PrepareSendResponse, PaymentError> {
         self.ensure_is_started().await?;
 
-        self.ensure_send_is_not_self_transfer(&req.invoice)?;
-        let invoice = self.validate_invoice(&req.invoice)?;
+        let fees_sat;
+        let receiver_amount_sat;
+        let parsed_destination = sdk_common::input_parser::parse(&req.destination).await?;
+        match &parsed_destination {
+            InputType::LiquidAddress {
+                address: liquid_address_data,
+            } => {
+                let Some(amount_sat) = liquid_address_data.amount_sat else {
```

Comment:
It seems that if the bip21 won't contain any amount (as it is optional) we will fail here.
Do we have to make it required?

Reply:
What is the expected flow? If the BIP21 doesn't contain any information on the amount we can't calculate fees, etc. which is what the prepare statement is primarily for.

Perhaps we could set the fees as optional and when no amount is given we can just return no fees and give the ability to set an amount in the `send_payment` method?

Reply:
I see. 
So perhaps we should pass tothe prepare_send_payment also the amount?